diff --git a/compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake b/compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake
index ad5b139..d79e85c 100644
--- a/compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake
+++ b/compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake
@@ -49,6 +49,7 @@ else()
 endif()
 
 set(ALL_GWP_ASAN_SUPPORTED_ARCH ${X86} ${X86_64} ${ARM32} ${ARM64} ${LOONGARCH64}) # OHOS_LOCAL
+set(ALL_OHOS_MEMGRAPH_SUPPORTED_ARCH ${ARM64})
 if(APPLE)
   set(ALL_LSAN_SUPPORTED_ARCH ${X86} ${X86_64} ${MIPS64} ${ARM64})
 else()
diff --git a/compiler-rt/cmake/config-ix.cmake b/compiler-rt/cmake/config-ix.cmake
index c0946af..bbb0b07 100644
--- a/compiler-rt/cmake/config-ix.cmake
+++ b/compiler-rt/cmake/config-ix.cmake
@@ -637,6 +637,9 @@ if(APPLE)
   list_intersect(ORC_SUPPORTED_ARCH
     ALL_ORC_SUPPORTED_ARCH
     SANITIZER_COMMON_SUPPORTED_ARCH)
+  list_intersect(OHOS_MEMGRAPH_SUPPORTED_ARCH
+    ALL_OHOS_MEMGRAPH_SUPPORTED_ARCH
+    SANITIZER_COMMON_SUPPORTED_ARCH)
 
 else()
   # Architectures supported by compiler-rt libraries.
@@ -667,6 +670,7 @@ else()
   filter_available_targets(SHADOWCALLSTACK_SUPPORTED_ARCH
     ${ALL_SHADOWCALLSTACK_SUPPORTED_ARCH})
   filter_available_targets(GWP_ASAN_SUPPORTED_ARCH ${ALL_GWP_ASAN_SUPPORTED_ARCH})
+  filter_available_targets(OHOS_MEMGRAPH_SUPPORTED_ARCH ${ALL_OHOS_MEMGRAPH_SUPPORTED_ARCH})
   filter_available_targets(ORC_SUPPORTED_ARCH ${ALL_ORC_SUPPORTED_ARCH})
 endif()
 
@@ -879,3 +883,11 @@ endif()
 # OHOS_LOCAL end
 pythonize_bool(COMPILER_RT_HAS_GWP_ASAN)
 
+# OHOS memgraph runtime: only build for OHOS on supported targets.
+if (OHOS_MEMGRAPH_SUPPORTED_ARCH AND COMPILER_RT_HAS_SANITIZER_COMMON AND
+    OS_NAME MATCHES "OHOS")
+  set(COMPILER_RT_HAS_OHOS_MEMGRAPH TRUE)
+else()
+  set(COMPILER_RT_HAS_OHOS_MEMGRAPH FALSE)
+endif()
+pythonize_bool(COMPILER_RT_HAS_OHOS_MEMGRAPH)
diff --git a/compiler-rt/lib/CMakeLists.txt b/compiler-rt/lib/CMakeLists.txt
index 18eed24..298b0f8 100644
--- a/compiler-rt/lib/CMakeLists.txt
+++ b/compiler-rt/lib/CMakeLists.txt
@@ -36,6 +36,10 @@ endfunction()
 
 if(COMPILER_RT_BUILD_SANITIZERS OR COMPILER_RT_BUILD_MEMPROF)
   compiler_rt_build_runtime(interception)
+
+  if(COMPILER_RT_HAS_OHOS_MEMGRAPH)
+    add_subdirectory(ohos_memgraph)
+  endif()
 endif()
 
 if(COMPILER_RT_BUILD_SANITIZERS)
