diff --git a/config/clang/clang.gni b/config/clang/clang.gni
index 9f751da7..8af686bb 100755
--- a/config/clang/clang.gni
+++ b/config/clang/clang.gni
@@ -11,8 +11,10 @@ if (!is_llvm_build) {
   default_clang_base_path = "${toolchains_dir}/${host_platform_dir}/llvm"
   clang_lib_path = "${toolchains_dir}/${host_platform_dir}/llvm/lib/clang/${clang_version}/lib"
 } else {
-  default_clang_base_path = "//out/llvm-install"
-  clang_lib_path = "//out/llvm-install/lib/clang/${clang_version}/lib"
+  #default_clang_base_path = "//out/llvm-install"
+  #clang_lib_path = "//out/llvm-install/lib/clang/${clang_version}/lib"
+  default_clang_base_path = "//prebuilts/clang/ohos/linux-x86_64/clang-15.0.4"
+  clang_lib_path = "//prebuilts/clang/ohos/linux-x86_64/clang-15.0.4/lib/clang/15.0.4/lib"
 }
 
 if (target_cpu == "arm") {
diff --git a/toolchain/toolchain.gni b/toolchain/toolchain.gni
index cfdca34c..ce101d63 100755
--- a/toolchain/toolchain.gni
+++ b/toolchain/toolchain.gni
@@ -51,7 +51,7 @@ hermetic_xcode_path =
 declare_args() {
   if (is_clang) {
     # Clang compiler version. Clang files are placed at version-dependent paths.
-    clang_version = "15.0.4"
+    clang_version = "16"
   }
   use_custom_clang = true
 }
